de/tensor.py

```diff
@@ -56,6 +56,19 @@ def form_tensor() -> np.ndarray:
 
     return Tensor, gene_names, cellLines
 
+def factorize(num_comp=6):
+    """ Using Parafac as a tensor factorization. """
+    tensor, genes, cellLines = form_tensor()
+    # perform parafac and CP decomposition
+    r2x_parafac = []
+    for i in range(1, 7):
+        # parafac
+        fac_p = parafac(tensor, rank=i, svd="randomized_svd")
+        r2x_parafac.append(1 - ((tl.norm(tl.cp_to_tensor(fac_p) - tensor) ** 2) / tl.norm(tensor) ** 2))
+
+    tfac = fac_p = parafac(tensor, rank=num_comp, svd="randomized_svd")
```

Comment:
Why do you have `tfac` and `fac_p`?

Reply:
Also, you should just save the factorization from above, so that you don't have to rerun it here.

Reply:
that was a typo. I fixed it. 
Saved the factorization in the new commit.